src/particles/ImpactXParticleContainer.cpp

```diff
@@ -242,28 +242,40 @@ namespace impactx
         amrex::Gpu::DeviceVector<amrex::ParticleReal> const & pt,
         amrex::ParticleReal qm,
         std::optional<amrex::ParticleReal> bunch_charge,
-        std::optional<amrex::Gpu::DeviceVector<amrex::ParticleReal>> w
+        std::optional<amrex::Gpu::DeviceVector<amrex::ParticleReal>> w,
+        std::optional<amrex::Gpu::DeviceVector<amrex::ParticleReal>> sx,
```

Comment:
@ax3l Ah - lines 245-246 appear to pose a problem.  How does the compiler know whether the first argument after `bunch_charge` is the (optional) vector `w` or the (optional) vector `sx`?

Reply:
Got it; easy fix.